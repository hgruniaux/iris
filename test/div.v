fn f(a, b) {
  return a + b * 2;
}

fn main() {
  let x = f(3, 4); // 3 + 4 * 2 = 11
  print x;
  let y = 3;
  print y;
  let z = x / y;
  let w = x % y;
  print z;
  print w;

  // Power of twos
  print(x / 8);
  print(x % 8);

  // Algebraic identities
  print(x / 1);
  print(x / x);
  print(x % 1);
  print(x % x);

  return 0;
}
