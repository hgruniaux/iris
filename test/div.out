11
3
3
2
1
3
11
1
0
0
